### DISCUSSION TITLE
Understanding violations

### DISCUSSION THREAD

**Discussion by neelkrish** [microsoft/sca-fuzzer#52] [Q&A] [answered]
Is there a documentation on how to process, debug and understand what violations I might be seeing after running the fuzzer?

**Comment by OleksiiOleksenko** [comment]
We don't but it's a very good suggestion. I'll try to write a brief tutorial next week.

**Reply by OleksiiOleksenko** [reply]
@neelkrish I pushed a [small guide](https://github.com/microsoft/sca-fuzzer/blob/main/docs/fuzzing-guide.md) for minimizing a violation and finding the speculative instruction. The guide does not yet go all the way to reverse-engineering the source of speculation, but it should be already enough for simple cases. Let me know if it is helpful and if you have any more questions.

I'll try to write the rest of the guide next week.

**Comment by OleksiiOleksenko** [comment]
A complete guide is now available at: https://microsoft.github.io/sca-fuzzer/user/fuzzing-guide/

**Reply by OleksiiOleksenko** [reply, marked as answer]
Moved to: https://microsoft.github.io/side-channel-fuzzer/howto/root-cause-a-violation/
